kubernetes/deployments_kubeclient.go

```diff
@@ -895,6 +900,10 @@ func (kc *kubeClient) deleteDeploymentConfig(spaceName string, appName string, n
 	// Delete all dependent objects and then this DC
 	policy := metaV1.DeletePropagationForeground
 	opts := &metaV1.DeleteOptions{
+		TypeMeta: metaV1.TypeMeta{ // Normally set automatically by k8s client-go
+			Kind:       "DeleteOptions",
+			APIVersion: "v1",
+		},
```

Comment:
This looks like it's either old code, or belongs in a separate PR.

Reply:
Ah, I forgot to mention this in the description. This was something caught when refactoring the tests. The Kubernetes client-go JSON serializer fills in the missing Kind/APIVersion fields so you don't have to do it every time. When sending the HTTP requests ourselves using json.Marshall, this doesn't happen. The OpenShift API server appears to accept the requests anyway, but I thought it would be best to fill in these fields.

Reply:
It's definitely more correct this way, if k8s depends on the APIVersion field instead of a separate endpoint to provide versioning.

Reply:
They do include the version in the endpoint, but also include it in every resource. It sounds like it should definitely be specified in the resource.

From https://github.com/kubernetes/community/blob/master/contributors/devel/api-conventions.md:
> All of the JSON accepted and returned by the server has a schema, identified by the "kind" and "apiVersion" fields. Where relevant HTTP header fields exist, they should mirror the content of JSON fields, but the information should not be represented only in the HTTP header.